### PR TITLE
[DEV-3478] add cfda column to assistance transaction tables

### PR DESCRIPTION
**Pull request by emily-brents**
**High level description:**

- Adds row containing CFDA Number to the transactions table on the award summary pages
- Tightens up horizontal spacing in transactions table

**Technical details:**
- Added CFDA number to CoreTransaction and BaseAssistanceTransaction
- Switches between assistance and contract (and loan) tables in TransactionTableContainer
- Switches between contract table mapping and assistance table mapping in TransactionTableHeaderCell when the contract table mapping returns null. 

**JIRA Ticket:**
[DEV-3478](https://federal-spending-transparency.atlassian.net/browse/DEV-3478)

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [ ] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [X] Verified cross-browser compatibility
- [X] Verified mobile/tablet/desktop/monitor responsiveness
- [X] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [x] Added Unit Tests for methods in Container Components, reducers, and helper functions `if applicable`
- [ ] All componentWillReceiveProps, componentWillMount, and componentWillUpdate in relevant child/parent components/containers replaced with [future compatible life-cycle methods](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html) per [JIRA item 3339](https://federal-spending-transparency.atlassian.net/browse/DEV-3339)
- [X] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated
- [ ] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [X] Design review complete
- [x] [API #2340](https://github.com/fedspendingtransparency/usaspending-api/pull/2340) merged concurrently
- [x] Code review complete

### CODE REVIEW

File: src/js/models/v2/awards/transactions/CoreTransaction.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ const CoreTransaction = {
         this._actionTypeDescription = data.actionTypeDescription || '';
         this.modificationNumber = data.modificationNumber || '';
         this.description = data.description || '';
+        this.cfdaNumber = data.cfda_number || '';
```
Comment (MaxwellKendall) [inline src/js/models/v2/awards/transactions/CoreTransaction.js]: Could you add a test for this?

File: src/js/components/award/table/TransactionsTable.jsx [hunk 2]
```diff
@@ -110,7 +110,7 @@ export default class TransactionsTable extends React.Component {
             const isLast = i === tableMapping.table._order.length - 1;
 
             const displayName = tableMapping.table[column];
-            let columnWidth = Math.max(measureTableHeader(displayName),
+            let columnWidth = Math.max(measureTableHeader(displayName) - 40,
```
Comment (MaxwellKendall) [inline src/js/components/award/table/TransactionsTable.jsx]: recommend defining a variable outside the definition of the class called `columnWidthOffset` and setting it to 40. That way it's a little easier to read.

**Comment by jonhill13** [comment]
agreed with @MaxwellKendall 
Screen Shot 2020-03-16 at 12 12 36 PM

File: src/js/components/award/table/TransactionsTable.jsx (hunk 2 above)
Comment (emily-brents) [reply src/js/components/award/table/TransactionsTable.jsx]: fixed

File: src/js/models/v2/awards/transactions/CoreTransaction.js (hunk 1 above)
Comment (emily-brents) [reply src/js/models/v2/awards/transactions/CoreTransaction.js]: Sure. Test was added in CoreTransaction-test.js.

**Comment by emily-brents** [comment]
> agreed with @MaxwellKendall
> 
@jonhill13 I see what the problem is in the screenshot, but that happens on all environments, so it doesn't seem to be an issue with the new code changes.

**Comment by jonhill13** [comment]
@emily-brents  from that logic agreed.

**Comment by MaxwellKendall** [comment]
@emily-brents @jonhill13 can we open a bug for that ss? Or does one exist?

**Comment by MaxwellKendall** [comment]
@emily-brents merge wheneva -- there's still a few checkboxes unchecked so I defer to you to merge when this is ready.

**Pull request merged**